static/app/views/profiling/profileDetails/components/profileDetailsTable.tsx

```diff
@@ -0,0 +1,477 @@
+import {useEffectAfterFirstRender} from 'sentry/utils/useEffectAfterFirstRender';
+import {useLocation} from 'sentry/utils/useLocation';
+import {useParams} from 'sentry/utils/useParams';
+
+import {useProfileGroup} from '../../profileGroupProvider';
+import {useColumnFilters} from '../hooks/useColumnFilters';
+import {useFuseSearch} from '../hooks/useFuseSearch';
+import {usePageLinks} from '../hooks/usePageLinks';
+import {useQuerystringState} from '../hooks/useQuerystringState';
+import {useSortableColumns} from '../hooks/useSortableColumn';
+import {aggregate, AggregateColumnConfig, collectProfileFrames} from '../utils';
+
+const RESULTS_PER_PAGE = 50;
+
+export function ProfileDetailsTable() {
+  const location = useLocation();
+  const [state] = useProfileGroup();
+  const [groupByViewKey, setGroupByView] = useQuerystringState({
+    key: 'detailView',
+    defaultState: 'occurrence',
+  });
+
+  const [searchQuery, setSearchQuery] = useQuerystringState({
+    key: 'query',
+    defaultState: '',
+  });
+
+  const [paginationCursor, setPaginationCursor] = useQuerystringState({
+    key: 'cursor',
+    defaultState: '',
+  });
+
+  const groupByView = GROUP_BY_OPTIONS[groupByViewKey] ?? GROUP_BY_OPTIONS.occurrence;
+
+  const cursor = parseInt(paginationCursor, 10) || 0;
+
+  const allData: TableDataRow[] = useMemo(() => {
+    const data =
+      state.type === 'resolved' ? state.data.profiles.flatMap(collectProfileFrames) : [];
+
+    return groupByView.transform(data);
+  }, [state, groupByView]);
+
+  const {search} = useFuseSearch(allData, {
+    keys: groupByView.search.key,
+    threshold: 0.3,
+  });
+
+  const debouncedSearch = useMemo(
+    () => debounce(searchString => setFilteredDataBySearch(search(searchString)), 500),
+    [search]
+  );
+
+  const [filteredDataBySearch, setFilteredDataBySearch] = useState<TableDataRow[]>(() => {
+    return search(searchQuery);
+  });
+
+  const pageLinks = usePageLinks(filteredDataBySearch, cursor);
+
+  const {filters, columnFilters, filterPredicate} = useColumnFilters(allData, [
+    'type',
+    'image',
+  ]);
+
+  const {currentSort, generateSortLink, sortCompareFn} = useSortableColumns({
+    ...groupByView.sort,
+    querystringKey: 'functionsSort',
+  });
+
+  const handleSearch = useCallback(
+    searchString => {
+      setSearchQuery(searchString);
+      setPaginationCursor(undefined);
+      debouncedSearch(searchString);
+    },
+    [setPaginationCursor, setSearchQuery, debouncedSearch]
+  );
+
+  useEffectAfterFirstRender(() => {
+    setFilteredDataBySearch(search(searchQuery));
+    // purposely omitted `searchQuery` as we only want this to run once.
+    // future search filters are called by handleSearch
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [allData, search]);
+
+  const data = filteredDataBySearch
+    .filter(filterPredicate)
+    .sort(sortCompareFn)
+    .slice(cursor, cursor + RESULTS_PER_PAGE);
```

Comment:
Should this be inside a `useMemo`?

Reply:
i don't think this would be much of an optimization as anytime `filterPredicate` or `sortCompareFn` or `cursor` changes we'd need to rerun the `useMemo`.

with the way the component is laid out,  those user interactions are all that can be done here, and would cause that `useMemo` to run everytime anyway.


we _might_ get a bit of benefit if we break each mutation into it's on `useMemo`, i just don't see it being a huge deal yet.

i can play around and see if it makes that big of an issue; typically if we had a perf issue with an array mutation and excessive rendering i'd expect to see some jank, but that's not the case

Reply:
ended up breaking each mutation into a `useMemo` as i had mentioned above